pkg/controllers/binding/common.go

```diff
@@ -54,34 +55,54 @@ var workPredicateFn = builder.WithPredicates(predicate.Funcs{
 })
 
 // ensureWork ensure Work to be created or updated.
+// nolint:gocyclo
 func ensureWork(
 	c client.Client, resourceInterpreter resourceinterpreter.ResourceInterpreter, workload *unstructured.Unstructured,
 	overrideManager overridemanager.OverrideManager, binding metav1.Object, scope apiextensionsv1.ResourceScope,
 ) error {
-	var targetClusters []workv1alpha2.TargetCluster
-	var requiredByBindingSnapshot []workv1alpha2.BindingSnapshot
+	var (
+		targetClusters            []workv1alpha2.TargetCluster
+		requiredByBindingSnapshot []workv1alpha2.BindingSnapshot
+		needReviseReplica         bool
+		desireReplicaInfos        map[string]int64
+		err                       error
+	)
 	switch scope {
 	case apiextensionsv1.NamespaceScoped:
 		bindingObj := binding.(*workv1alpha2.ResourceBinding)
 		targetClusters = bindingObj.Spec.Clusters
 		requiredByBindingSnapshot = bindingObj.Spec.RequiredBy
+		needReviseReplica = bindingObj.Spec.Replicas > 0
 	case apiextensionsv1.ClusterScoped:
 		bindingObj := binding.(*workv1alpha2.ClusterResourceBinding)
 		targetClusters = bindingObj.Spec.Clusters
 		requiredByBindingSnapshot = bindingObj.Spec.RequiredBy
+		needReviseReplica = bindingObj.Spec.Replicas > 0
 	}
 
 	targetClusters = mergeTargetClusters(targetClusters, requiredByBindingSnapshot)
+	if len(targetClusters) == 0 {
+		return nil
+	}
+
+	// If binding has replicas, those who have ReplicaScheduling policy should be revised.
+	if needReviseReplica {
+		needReviseReplica, err = hasReplicaSchedulingPolicy(binding.GetAnnotations())
+		if err != nil {
+			return err
+		}
+		if needReviseReplica {
+			desireReplicaInfos = transScheduleResultToMap(targetClusters)
+		}
+	}
```

Comment:
I think we can simply remove this logic and determine whether to continue the execution based on whether the rb scheduling is complete:

https://github.com/karmada-io/karmada/blob/c4428d6f224b432a47bfa6aa2bdd6b3a1ccfa275/pkg/apis/work/v1alpha2/binding_types.go#L222-L226

And `ResourceInterpreter.HookEnabled` will determine whether to revise.

Reply:
> I think we can simply remove this logic and determine whether to continue the execution based on whether the rb scheduling is complete.

By determining `SchedulerObservedGeneration` > 0 ? I'm afraid if you delete `applied-placement` manually(which I have tried this trick before🤣 to fully reschedule a deployment), there could be some problems here.

Reply:
> By determining SchedulerObservedGeneration > 0 ?

By determining `rb.Status.SchedulerObservedGeneration == rb.Generation`, this indicates that the current rb is the latest result processed by the scheduler.

If this method is used for judgment, we need to change the previous usage habit. For the pp whose `ReplicaScheduling` is nil, and the matched resource is workload, the replicas field in `Spec.Clusters` needs to be filled scheduling is complete.

Reply:
1. If `rb.Status.SchedulerObservedGeneration != rb.Generation`, just return nil.
2. And then if `ReplicaScheduling` is nil, we don't revise replicas.

Right?

Reply:
Just the first point is enough, and it's up to `enable` function to decide whether to revise.

Reply:
I don't think it gonna works. If `ReplicaScheduling` is nil, scheduler consider this workload as a kind of 'common' resources, like configMap or secret, so the result could be `[member1, member2, member3]` without replica assigned. But here if we don't care about `ReplicaScheduling`and let `enable` function to decide, replicas with all member clusters will be zero by revision which cause an expected result.

Reply:
Oh, I think we have conclued a perfect solution in private😄.